frontend/catalyst/api_extensions/differentiation.py

```diff
@@ -790,7 +793,7 @@ def _ensure_differentiable(f: DifferentiableLike) -> Differentiable:
     elif isinstance(f, Callable):  # Keep at the bottom
         return Function(f)
 
-    raise DifferentiableCompileError(f"Non-differentiable object passed: {type(f)}")
+    raise DifferentiableCompileError(f"Differentiation target must be callable, got: {type(f)}")
```

Comment:
Is it not just a type error?

Reply:
Yeah that's probably appropriate too, Sergei introduced this error to indicate failures during compilation specifically related to differentiation. Happy to change it though.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1190/commits/ab99e7c31492fbbad96ed831c06acdd790237933